File: star_pattern/StarPattern.cs
using System;
using System.Linq;

public static class StarPattern
{
    public static void Main(string[] args)
    {
        int iterations = 0;
        int starCount = 0;
        int size = 15;
        while (iterations < size)
        {
            iterations++;
            if (iterations <= (size / 2))
            {
                i++;
                Console.Write(string.Concat(Enumerable.Repeat(" ", size - starCount)));
                Console.WriteLine(string.Concat(Enumerable.Repeat("* ", i)));

            }
            else
            {
                i--;
                Console.Write(string.Concat(Enumerable.Repeat(" ", size - starCount)));
                Console.WriteLine(string.Concat(Enumerable.Repeat("* ", i)));
            }
        }
    }
}

File: star_pattern/star_pattern.lua
iterations = 0
i = 0
size = 15
while (iterations < size)
do
    iterations = iterations + 1
    if (iterations <= size // 2)
    then
        i = i + 1
        io.write(string.rep(" ", (size - i)))
        print(string.rep("* ", i))
    else
        i = i - 1
        io.write(string.rep(" ", (size - i)))
        print(string.rep("* ", i))
    end
end